src/XMakeBuildEngine/Construction/ProjectElementContainer.cs

```diff
@@ -630,10 +620,7 @@ private void AddInitialChild(ProjectElement child)
 
             _count++;
 
-            if (!child.IsImplicit)
-            {
-                MarkDirty("Add child element named '{0}'", child.ElementName);
-            }
+            MarkDirty("Add child element named '{0}'", child.ElementName);
```

Comment:
This looks like a change in behavior. Does it need to happen for Remove just above this? Or others?